### PR TITLE
Views

### PR DESCRIPTION
**Pull request by wuyepabdul**
In this milestone, the following was achieved:

- Implemented design from sneak wireframe
-

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        <%= link_to user_post_path(@user.id, post.id) do %>
+            <div class='post-div'>
+                <h4> Post 1</h4>
+                <p> <%= post.text %> </p>
+            </div>
+        <% end %>
+        <div class='comment-div'>
+        <% post.comments.each do |comment| %>
+            <p> @user.name: <%= comment.text %> </p>
+        <% end %>
```
Comment (Lameck1) [inline app/views/posts/index.html.erb]: - [ ] The comments we need to render here are not all the comments. It should be the 5 most recent comments according to the wireframe. Please refactor `post.recent_comments.each`.

- [ ] In the next line where we have `<p> @user.name: <%= comment.text %> </p>` we are expected to render the names of those who left comments on the post. Please look at the example below and refactor code as necessary.
  ```ruby
  <p> <%= comment.author.name %>: <%= comment.text %> </p>
  ```

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <div class='comment-div'>
+        <p> Username: Comment 1 </p>
+        <p> Username: Comment 2 </p>
+        <p> Username: Comment 3 </p>
+        <p> Username: Comment 5 </p>
+        <p> Username: Comment 6 </p>
+        <p> Username: Comment 7 </p>
+        <p> Username: Comment 8 </p>
+    </div>
+</div>
```
Comment (Lameck1) [inline app/views/posts/show.html.erb]: - [ ] This should have been generated by code and not hardcoded.  This view for showing details of a single post should work together with the show action of the posts_controller which is currently not implemented. The show action is talked about in the next thread.

### Example code that should be in this view:

```ruby
<div class='post-comment-wrapper'>
    <div class='post-div'>
        <h4><%= @post.title %></h4>
        <p> <%= @post.text %> </p>
    </div>
    <div class='comment-div'>
        <% @post.comments.each do |comment| %>
            <p> <%= comment.author.name %>: <%= comment.text %> </p>
        <% end %>
    </div>
</div>
```

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class PostsController < ApplicationController
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.recent_posts
+  end
+
+  def show; end
```
Comment (Lameck1) [inline app/controllers/posts_controller.rb]: - [ ] In line with the above suggestion, this show action needs implementation for us to be able to display a single post. Please consider the suggestion below.
```suggestion
  def show
    @user = User.find(params[:user_id])
    @post = @user.posts.find(params[:id])
  end
```

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        <div class='post-div'>
+            <h4> <%= post.title %> </h4>
+            <p> lorem ipsum deprocadi ips deasdflorem ipsum deprocadi ips deasdflorem ipsum deprocadi ips deasdflorem ipsum deprocadi ips deasdflorem ipsumdeasdflorem ipsum deprocadi ips deasdflorem ipsum ... </p>
+        </div>
+    <% end %>
+<% end %>
+
+
+<div class='button'>
+    <button> See all Posts </button>
```
Comment (Lameck1) [inline app/views/users/show.html.erb:29]: - [ ] This button(**See all Post**) should link to user's post index as displayed on the [wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_all_posts.png?token=ABIFLA3SVPWT6MXQ3C2OGFLBHIHES). Please make it a link to all user's posts.

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    <h4> Bio</h4>
+    <p> <%= @user.bio %> </p>
+
+</div>
+
+<% @posts.each do |post| %>
+    <%= link_to user_posts_path(@user.id) do %>
+        <div class='post-div'>
+            <h4> <%= post.title %> </h4>
+            <p> lorem ipsum deprocadi ips deasdflorem ipsum deprocadi ips deasdflorem ipsum deprocadi ips deasdflorem ipsum deprocadi ips deasdflorem ipsumdeasdflorem ipsum deprocadi ips deasdflorem ipsum ... </p>
```
Comment (Lameck1) [inline app/views/users/show.html.erb]: - [ ] Let's not hard code this. It should be the actual text from the post. Just like you've displayed `post.text`, please also display `post.text` in this `<p>` tag.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi, @wuyepabdul, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
Hi @wuyepabdul, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by wuyepabdul** [comment]
Thanks for the review @Lameck1 and @uwadonat. Your suggestions were really helpful to me

**Pull request merged**